pkg/operator/resources/cluster/factory.go

```diff
@@ -39,6 +39,9 @@ func createCRDResources(args *FactoryArgs) []runtime.Object {
 	return []runtime.Object{
 		createDataVolumeCRD(),
 		createCDIConfigCRD(),
+		createVolumeSnapshotClassCRD(),
```

Comment:
Do we really want to be creating these CRDs if they don't already exist?  I don't think so.  We can detect if the VolumeSnapshot api does not exist and always do host assisted cloning.

Reply:
What about the case that CDI is installed before the CSI driver?

Reply:
The smart-cloner-controller is registered in the cmd/controller. I.e. we listen to the snapshot informer events after deployment. So if we don't create the CRDs, it means that we must rely on them being already installed before deployment. Thus, if not installed, we need to stop the smart-cloner-controller? So we'll have to detect an installation of the CRDs afterwards to run the controller again?

Reply:
Can't we have the operator (or deployment) controller listen for CRDs being created, and if it finds the right CRDs to create smart cloner controller, or if the CRDs are removed it removes the smart cloner controller.

Reply:
Yeah, I think the operator should watch CRDs.  If VolumeSnapshot exists either a dedicated deployment for the smart clone controller will be started or the current controller deployment can be started/restarted with a different set of args

Reply:
Sounds like a major change in the operator behaviour? Can we do it in a consecutive PR later on and just manually create the CRDs for now?